git-coco

```diff
@@ -164,6 +164,47 @@ pr() {
     echo "https://console.aws.amazon.com/codesuite/codecommit/repositories/${repo}/pull-requests/${id}/changes"
 }
 
+prlist() {
+    local repo list_of_prs pull_info
+    local dst_branch src_branch is_merge pull_id author title
+
+    repo="$1"
+
+    if [ -z "$repo" ]; then
+        fail "Usage: git coco prlist <repo>"
+    fi
```

Comment:
What would be super cool is if this could check `git remote -v` and determine the repo if one isn't specified

There are 4 cases to cover:

1. `codecommit:` with no profile, just repo name
2. `codecommit:` with profile and repo name
3. `https:` that resolves to CodeCommit
4. Other

Example of 1:

```
origin	codecommit::us-east-1://Test1 (fetch)
origin	codecommit::us-east-1://Test1 (push)
```

Example of 2:

```
origin	codecommit::us-east-1://coco@Test2 (fetch)
origin	codecommit::us-east-1://coco@Test2 (push)
```

Example of 3:

```
origin	https://git-codecommit.us-east-1.amazonaws.com/v1/repos/Test3 (fetch)
origin	https://git-codecommit.us-east-1.amazonaws.com/v1/repos/Test3 (push)
```

Example of 4:

```
origin	https://github.com/rkeene/git-coco.git (fetch)
origin	https://github.com/rkeene/git-coco.git (push)
```

If the repo (and possibly profile) can be found from one of these patterns, then we could use that information to query AWS CodeCommit.

Reply:
Added a url_parser() function to parse urls passed to it. Function sets _profile variable if a profile is found from a codecommit:// url. In all other cases it returns (via echo) the repository name parsed from the url.

prlist() calls url_parser() with the output of git remote -v passed in to be parsed.

Reply:
This looks cool, I added some more comments around it above.

Thanks !  :+1: